### ISSUE TITLE
Add `pygments` to extras

### ISSUE THREAD

**Issue by Qu4tro** [django-extensions/django-extensions#1468]
Hello there :wave: 

`pygments`  is currently used to color the SQL output and it's only used when it's installed. That's great!
On the other hand, since it doesn't explicitly require it, it leaves people depending on `django-extensions` to guess about which `pygments` are compatible with it.

My opinion is that it might be worth it, to add `pygments` to the `extras_requires` in `setup.py` . 

Lmk what do you think.

**Comment by trbs** [comment]
I think this makes sense.

Also I guess this should also be better explained in the docs...


Are you also thinking of completing the list for other optional libraries ? like the ones from `shell_plus` or 'export_email' ?

**Comment by Qu4tro** [comment]
I think it would be a plus to have all optional requirements as extras.

From a quick we have 
- `requests` 
    - `django_extensions/management/commands/pipchecker.py`
- `MySQLdb`:
   - `django_extensions/management/commands/reset_db.py` 
   - `django_extensions/management/commands/drop_test_database.py`
- `psycopg2` 
   - `django_extensions/management/commands/reset_db.py` 
   - `django_extensions/management/commands/drop_test_database.py`
- `boto` 
   - `django_extensions/management/commands/sync_s3.py` 
- `vobject` 
   - `django_extensions/management/commands/export_emails.py` 
- `pyyaml` 
   - `django_extensions/management/commands/print_settings.py` 
- `shortuuid` 
   - `django_extensions/db/fields/__init__.py`
- `pygments`
   - `django_extensions/management/debug_cursor.py`
- `sqlparse`
   - `django_extensions/management/debug_cursor.py`

- `ipdb`
   - `django_extensions/management/debug_cursor.py`
- `iPython`
   - `django_extensions/management/debug_cursor.py`
- `pydot`
   - `django_extensions/management/graph_models.py`
- `pydotplus`
   - `django_extensions/management/graph_models.py`
- `pygraphviz`
   - `django_extensions/management/graph_models.py`
- `hotshot`
   - `django_extensions/management/commands/runprofileserver.py`

Does this sound right to you?

There might be an issue finding the correct versions for `django-extensions`. 

This one-liner might help people share their versions here:
`grep -i -E -- '^(django-extensions|hotshot|pygraphviz|pydotplus|pydot|ipython|ipdb|sqlparse|pygments|shortuuid|pyyaml|vobject|boto|psycopg2|mysqldb|requests)=' <(pip freeze)`

And my output:
```
django-extensions==2.2.5
ipdb==0.12.2
ipython==7.9.0
psycopg2==2.8.4
Pygments==2.4.2
PyYAML==5.1.2
requests==2.22.0
sqlparse==0.3.0
```

**Comment by trbs** [comment]
List looks good.
`shell_plus` got a couple itself: `ptpython`, `notebook` / `jupyter`, `bpython` :-)

Think we do support much older versions of most of these libraries. 
I guess the easiest strategy would be to either not pin them at all (unless we know for sure we need a certain minimal version) or pin everything to >= current versions.

What I am wondering about is how to name the extras ?

Name them the same as the dependencies:
```
pip install django-extensions[psycopg2,ipython,requests]
``` 

Or by the command name:
```
pip install django-extensions[shell_plus,reset_db]
```

(but then in most cases you probably don't need both `psycopg2` and `mysqldb` installed...)

Or something else ?

My list in my default venv is:
```
boto==2.49.0
ipdb==0.12.3
ipython==7.9.0
psycopg2==2.8.4
Pygments==2.4.2
pygraphviz==1.5
PyYAML==5.1.1
requests==2.22.0
shortuuid==0.5.0
sqlparse==0.3.0
vobject==0.9.6.1
```

**Comment by Qu4tro** [comment]
> I guess the easiest strategy would be to either not pin them at all (unless we know for sure we need a certain minimal version) or pin everything to >= current versions.

I'm okay with not pinning them for now. We could add version constraints later on, if they become a problem.

> What I am wondering about is how to name the extras ?

I like the `Name them the same as the dependencies` option. `django-extensions` is already focused for Python developers. Masking the name might be confusing IMO.

**Comment by trbs** [comment]
May I ask if you can spearhead a PR for this ?

**Comment by Qu4tro** [comment]
Yeah, of course. I'm temporarily relocating, but I'll see if I can find the time.

**Comment by adevore** [comment]
Mind if I pick this up? Getting the optional dependencies as extras has been on my wish list for a while.

**Comment by Qu4tro** [comment]
Please do. Thank you!

**Comment by adevore** [comment]
Great. A few questions.

First, naming convention. Here's my first idea on how to do that:

Commands that require must have dependency to run would use their name for an extra:

```toml
[project.optional-dependencies]
runserver_plus = ["Werkzeug"]
```

Then additional extras use `{command}-{subfeature}`, like so:

```toml
[project.optional-dependencies]
runserver_plus-ssl=["openssl>=???"]
shell_plus-ipython = ["ipython>=???"]
shell_plus-ptpython=["ptpython>=???"]
```

This may require some duplication.

Second question. I'm not sure where to set the lower bounds for dependencies. I could use a few criteria:

* First release before April 3, 2023. That was the release date of Django 4.2.0. ~3 years.
* First release that adds 3.10 support, the lowest Python version under support. Often poorly documented. Some packages that are not actively maintained do not officially support 3.10 (e.g. django-pdb). ~5 years
* Lowest release that the code supports. This isn't always apparent and is sometimes for ancient versions. Up to 15 years old.

Third question. This will involve combing over the code base. I noticed support for things like Werzkerg before version 0.9 (released 2013). Is this still desired? Or would putting together a list of possible deprecations while I'm wandering around be a good use of time?
